File: competition/Feb 14. 2021/Solution5677.py
class Solution5677(object):
    def countHomogenous(self, s):
        """
        :type s: str
        :rtype: int
        """
        array = []
        cur = '+'
        number = 0
        for i in range(len(s)):
            cur2 = s[i]
            if cur2 == cur:
                number += 1
            else:
                array.append(number)
                number = 1
                cur = cur2
        array.append(number)
        count = 0
        x = 1000000007
        for i in range(len(array)):
            mm = array[i]
            count += mm * (mm + 1) / 2
        return count % x

s = Solution5677()
print(s.countHomogenous("abbcccaa"))

File: competition/First Half 2021/Feb 14. 2021/Solution5677.java
import java.util.List;
import java.util.LinkedList;

class Solution5677 {
    public int countHomogenous(String s) {
    	if(s.length() == 0){
    		return 0;
    	}
    	int x =  1000000007;
    	int size = s.length();
    	char cur = s.charAt(0);
    	int number = 0;
    	List<Integer> list = new LinkedList<>();
    	for(int i = 0; i < s.length(); i++){
    		char cur2 = s.charAt(i);
    		if(cur2 != cur){
    			cur = cur2;
    			list.add(number);
    			number = 1;
    		}
    		else{
    			number += 1;
    		}
    	}
    	list.add(number);
    	long count = 0;
    	for(int i = 0; i < list.size(); i++){
    		int cc = list.get(i);
    		long mm = (cc + (long) 1) * cc / 2;
    		count += (mm) % x;
    	}
    	return (int) count % x;
    }

    public static void main(String[] args){
    	Solution5677 s = new Solution5677();
    	System.out.println(s.countHomogenous("aadadw"));
    }
}